faststreams/inputs.nim

```diff
@@ -224,56 +268,52 @@ proc readableNow*(s: InputStream): bool =
   getNewSpan s
   s.span.hasRunway
 
-template readableNow*(s: AsyncInputStream): bool =
-  readableNow InputStream(s)
-
-# TODO: The pure async interface should be moved in a separate module
```

Comment:
Well, this was explaining the better way to do it, but I guess we could try it in another PR.
It's possible to address the root problem about the async macro in Chronos as well, so this comment still carries useful information.

Reply:
yes, that is an approach I'm trying to push for in general: https://github.com/nim-lang/Nim/pull/18700

that said, the code puts the async members in the non-async `InputStream` class and exposes some "otherwise" sync functions that use the async members conditionally - this make's it difficult to .. do it in a module without breaking API